archunit/src/test/java/com/tngtech/archunit/core/domain/JavaClassTest.java

```diff
@@ -624,6 +624,31 @@ public void direct_dependencies_to_self_finds_correct_set_of_origin_types() {
                 .transform(toGuava(GET_ORIGIN_CLASS)).toSet();
     }
 
+    @Test
+    public void stubs_have_empty_dependencies_and_are_not_imported() {
+        class Element {
+        }
+        class DependsOnArray {
+            Element[] array;
+        }
+        JavaClass directlyImportedClass = new ClassFileImporter().importClasses(DependsOnArray.class).get(DependsOnArray.class);
+        JavaClass additionallyImportedClass = directlyImportedClass.getField("array").getRawType();  // Element[]
+        JavaClass stub = additionallyImportedClass.getComponentType();  // Element
```

Comment:
Does this mean that component types are not resolved automatically, even if `resolveMissingDependenciesFromClasspath` is set to `true`? Because if yes, then I think this should actually be fixed! (and maybe this test then written with deactivated resolution from classpath to easier get a stub class...)

Reply:
I'll check again on `resolveMissingDependenciesFromClasspath`, but I believe that my issue #397 was related to that.

Reply:
Ah `resolveMissingDependenciesFromClasspath=true` was the default anyways. So no, component types don't seem to get resolved automatically.

Reply:
I think this meanwhile is fixed, i.e. component types will be resolved transitively automatically

Reply:
Right. I had confused stubs with classes that are not fully imported.